src/java/org/apache/cassandra/db/compaction/CompactionTask.java

```diff
@@ -397,6 +397,7 @@ public static ImmutableCoordinatorLogOffsets getCoordinatorLogOffsets(Set<SSTabl
         ImmutableCoordinatorLogOffsets.Builder builder = new ImmutableCoordinatorLogOffsets.Builder();
         for (SSTableReader sstable : sstables)
             builder.addAll(sstable.getCoordinatorLogOffsets());
+        builder.purgeTransfers();
```

Comment:
So the idea here is that transfer IDs that are durably reconciled will never trigger reconciliation anyway...so they just don't need to be in the new post-compaction SSTables? Makes sense, although I wonder if a short comment to that effect would be helpful...or...rename to `purgeReconciledTransfers()`...or maybe by favorite...make `Predicate<ShortMutationId> pred` an argument for `purgeTransfers()` and pass `MutationTrackingService.instance::isDurablyReconciled`.